OnlineLibrary.API/Controllers/BookController.cs

```diff
@@ -0,0 +1,132 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using System;
+using OnlineLibrary.BLL.Interfaces;
+using OnlineLibrary.Common.Entities;
+using OnlineLibrary.Common.Helpers;
+using OnlineLibrary.API.Model;
+using AutoMapper;
+
+namespace OnlineLibrary.API.Controllers
+{
+    [Route("api/[controller]/[action]")]
+    [ApiController]
+    public class BookController : Controller
+    {
+        private readonly IBookService _bookService;
+
+        private readonly IAuthorService _authorService;
+
+        public BookController(IBookService iBook, IAuthorService iAuthor)
+        {
+            _bookService = iBook;
+            _authorService = iAuthor;
+        }
+
+        // GET: api/Book/GetAllBooks
+        [HttpGet]
+        public IActionResult GetAllBooks()
+        {
+            try
+            {
+                return Ok(_bookService.GetAllBooks());
+            }
+            catch (Exception e)
+            {
+                return NotFound(e.Message);
+            }
+        }
+
+        // GET: api/Book/GetBooksWithFilters
+        [HttpGet("{authorId}/{name}/{reservation}/{inArchieve}")]
+        public IActionResult GetBooksWithFilters(int? authorId, string name, int? reservation, int? inArchieve)
+        {
+            try
+            {
+                return Ok(_bookService.FilterBooks(authorId, name, reservation, inArchieve));
+            }
+            catch (Exception e)
+            {
+                return NotFound(e.Message);
+            }
+        }
+
+        // GET: api/Book/GetBookById/[id]
+        [HttpGet("{id}")]
+        public IActionResult GetBookById(int? id)
+        {
+            try
+            {
+                return Ok(_bookService.GetBookById(id));
+            }
+            catch (Exception e)
+            {
+                return NotFound(e.Message);
+            }
+        }
+
+        // POST:  api/Book/Create
+        [HttpPost]
+        public IActionResult Create([FromBody] CreateBook cBook)
+        {
+            try
+            {
+                // необходимо настроить добавление с тегами!
+                ExceptionHelper.Check<Exception>(cBook.Genre == null || !Enum.IsDefined(typeof(Genre), cBook.Genre), "Даного жарна не существует");
+                // Config mapper.
+                var config = new MapperConfiguration(cfg => cfg.CreateMap<CreateBook, Book>()
```

Comment:
move to dedicated file, and inject IMapper instead of explicitly creating new instance